### ISSUE TITLE
pdb fails on basic toga apps

### ISSUE THREAD

**Issue by eteq** [beeware/toga#1922]
### Describe the bug

The standard python debugger `pdb` trows a bit of a fit when I try to use it on a Toga app. My specific app is a bit more complicated than my steps to reproduce, but the core problem is the same as the minimal case I describe below.

`pdb` would be really helpful in learning toga because it prevents me from having to stick a bunch of `print(dir(...))` and so on into my app when I'm just trying to learn basic toga functionality.

### Steps to reproduce

1. Start from really any app, but the simplest is probably the [tutorial 0 app](https://toga.readthedocs.io/en/stable/tutorial/tutorial-0.html). 
2. Add a 1/0 somewhere in the business logic of the app (I put it in the `button_handler` function but really anywhere other than before `main_loop` is called seems to do it).
3. Start the app as `python -m pdb hello.py` (I realize it's supposed to me `python -m hello` but of course that's not compatible with `pdb`, and for these apps, at least, `python hello.py` works flawlessly.
4. That leads to a traceback:
```
Traceback (most recent call last):
  File "/usr/lib/python3.10/pdb.py", line 1732, in main
    pdb._runscript(mainpyfile)
  File "/usr/lib/python3.10/pdb.py", line 1592, in _runscript
    self.run(statement)
  File "/usr/lib/python3.10/bdb.py", line 597, in run
    exec(cmd, globals, locals)
  File "<string>", line 1, in <module>
  File "<redacted>/hello.py", line 25, in <module>
    main().main_loop()
  File "<redacted>/hello.py", line 21, in main
    return toga.App("First App", "org.beeware.helloworld", startup=build)
  File "<redacted>/venv/lib/python3.10/site-packages/toga/app.py", line 227, in __init__
    main_module_pkg = sys.modules["__main__"].__package__
AttributeError: module '__main__' has no attribute '__package__'

```

### Expected behavior

I expected it to give me a traceback where I put the `1/0` so I could explore live objects in the debugger to fix my buisness logic.

### Environment

- Operating System: Arch Linux
- Python version: 3.10
- Software versions:
  - Briefcase: N/A
  - Toga: 0.3.1

### Logs

```

```

**Comment by eteq** [comment]
After submitting this I realized using `breakpoint()` *does* trigger `pdb` sucessfully.  That said, there are cases where the standard `python -m pdb ...` functionality is better for debugging, so it would still be useful if this was possible.

**Comment by freakboy3742** [comment]
Agreed that it would be nice - but it's also going to be extremely difficult to implement. Toga apps are multithreaded, and involve a complex interaction with the event loop of the underlying operating system. Blocking debugger calls are always going to be problematic, and non-blocking thread handling will also be problematic. 

In some cases, it's also unclear where PDB would be triggering. For example, Toga *won't* (or *shouldn't*) crash if an event handler raises an exception. This is done as a safety catch. In most cases, it's preferable for an app to report an error but continue if a handler raises an exception; but this impedes the ability of PDB to detect and intercept exceptions when they occur.

In summary: I agree that it would be desirable to have a much better story for debugging Toga apps; however, it's more in the territory of a "complex feature enhancement" rather than a bug.
